assets/js/components/Dashboard/AlertsPage.tsx

```diff
@@ -1,11 +1,130 @@
+import FilterDropdown from "./FilterDropdown";
+import { Col, Container, Row } from "react-bootstrap";
+import { ArrowDown, ArrowUp } from "react-bootstrap-icons";
+import "../../../css/screenplay.scss";
+import {
+  MODES_AND_LINES,
+  SORT_LABELS,
+  SCREEN_TYPES,
+  STATUSES,
+} from "../../constants/constants";
+import { Alert } from "../../models/alert";
 import classNames from "classnames";
 
+type DirectionID = 0 | 1;
+
+const getAlertSortLabel = (sortDirection: DirectionID) => {
+  const sortLabels = SORT_LABELS["Alerts"];
+  return sortLabels[sortDirection];
+};
+
 interface Props {
+  alerts: Alert[];
   isVisible: boolean;
 }
 
 const AlertsPage: ComponentType<Props> = (props: Props) => {
+  const [alertSortDirection, setAlertSortDirection] = useState<DirectionID>(0);
+  const [alertModeLineFilterValue, setAlertModeLineFilterValue] = useState(
+    MODES_AND_LINES[0]
+  );
+  const [alertScreenTypeFilterValue, setAlertScreenTypeFilterValue] = useState(
+    SCREEN_TYPES[0]
+  );
+  const [alertStatusFilterValue, setAlertStatusFilterValue] = useState(
+    STATUSES[0]
+  );
+
+  const alertSortLabel = getAlertSortLabel(alertSortDirection);
+  const sortLabelOnClick = () => {
+    setAlertSortDirection((1 - alertSortDirection) as DirectionID);
+  };
+
+  const handleAlertModeOrLineSelect = (value: string) => {
+    const selectedFilter = MODES_AND_LINES.find(({ label }) => label === value);
+    if (selectedFilter) {
+      setAlertModeLineFilterValue(selectedFilter);
+    }
+  };
+
+  const handleAlertScreenTypeSelect = (value: string) => {
+    const selectedFilter = SCREEN_TYPES.find(({ label }) => label === value);
+    if (selectedFilter) {
+      setAlertScreenTypeFilterValue(selectedFilter);
+    }
+  };
+
+  const handleAlertStatusSelect = (value: string) => {
+    const selectedFilter = STATUSES.find(({ label }) => label === value);
+    if (selectedFilter) {
+      setAlertStatusFilterValue(selectedFilter);
+    }
+  };
+
+  const filterAlerts = () => {
+    let filteredAlerts = props.alerts;
+    if (alertModeLineFilterValue !== MODES_AND_LINES[0]) {
+      filteredAlerts = filterAlertsByModeOrLine(
+        filteredAlerts,
+        alertModeLineFilterValue
+      );
+    }
+    return filteredAlerts;
+  };
+
+  const filterAlertsByModeOrLine = (
+    alerts: Alert[],
+    { label, ids }: { label: string; ids: string[] }
+  ) => {
+    return alerts.filter((alert) => {
+      return alert.attributes.informed_entity.some((informedEntity) => {
+        switch (label) {
+          case "Commuter Rail":
+            return informedEntity.route_type === 2;
+          case "Bus":
+            return informedEntity.route_type === 3;
+          case "Ferry":
+            return informedEntity.route_type === 4;
+          default:
+            return ids.includes(informedEntity.route);
+        }
+      });
+    });
+  };
+
+  const filterAlertsByScreenType = (alerts: Alert[]) => {};
```

Comment:
This is on hold until we know what the data will look like once we're able to map alerts to screen types.